hallux/targets/gitlab_suggestion.py

```diff
@@ -0,0 +1,167 @@
+        FilesystemTarget.__init__(self)
+
+        if "GITLAB_TOKEN" not in os.environ.keys():
+            raise SystemError("GITLAB_TOKEN is not provided")
+
+        answ = GitlabSuggestion.parse_mr_url(mr_url)
+        if answ is None:
+            logger.warning(f"Cannot parse gitlab MR URL: {mr_url}")
+            raise SystemError(f"Cannot parse gitlab MR URL: {mr_url}")
+        (base_url, project_name, mr_iid) = answ
+        self.base_url: Final[str] = base_url
+        self.project_name: Final[str] = project_name
+        self.mr_iid: Final[int] = mr_iid
+
+        request_url: str = f"{base_url}/projects/{project_name}/merge_requests/{mr_iid}/changes?unidiff=true"
+        headers = {"PRIVATE-TOKEN": f"{os.environ['GITLAB_TOKEN']}"}
+        mr_response = requests.get(request_url, headers=headers)
+
+        if mr_response.status_code != 200:
+            logger.error(mr_response.text)
+            raise SystemError(mr_response.text)
+
+        self.mr_json = mr_response.json()
+        self.base_sha: Final[str] = self.mr_json["diff_refs"]["base_sha"]
+        self.head_sha: Final[str] = self.mr_json["diff_refs"]["head_sha"]
+        self.start_sha: Final[str] = self.mr_json["diff_refs"]["start_sha"]
+
+        # mapping from new_path to old_path (if file is changed)
+        self.changed_files: Final[dict[str, str]] = {}
+        # mapping for diffs of changed files
+        self.changed_diffs: Final[dict[str, str]] = {}
+        for change in self.mr_json["changes"]:
+            new_path = change["new_path"]
+            if new_path is not None:  # new_path is None when file is deleted
+                self.changed_files[new_path] = change["old_path"]
+                self.changed_diffs[new_path] = change["diff"]
+
+        if self.mr_json["merged_at"] is not None or self.mr_json["closed_at"] is not None:
+            raise SystemError(f"Merge Request {mr_iid} is either closed or merged already")
+
+        local_git_sha: str
+        try:
+            git_log: str = subprocess.check_output(["git", "log", "--pretty=oneline"]).decode("utf8")
+            local_git_sha = git_log.split("\n")[0].split(" ")[0]
+        except Exception as e:
+            raise SystemError(f"Local git commit is not available: {e}")
+
+        if local_git_sha != self.head_sha:
+            raise SystemError(
+                f"Local git commit: `{local_git_sha}` "
+                f"do not coinside with the head from pull-request: `{self.head_sha}` "
+            )
+
+    @staticmethod
+    def parse_mr_url(mr_url: str) -> tuple[str, str, int] | None:
+        """
+        Tries to parse Pull-Request URL to obtain base_url, repo_name and PR_ID
+        :param mr_url: shall look like this: https://gitlab.com/hallux/hallux/-/merge_requests/2
+        :return: (base_url, project_name, mr_iid) OR (None, None, None)
+        """
+        if mr_url.startswith("https://"):
+            mr_url = mr_url[len("https://") :]
```

Comment:
Define a constant instead of duplicating this literal "https://" 3 times.
```suggestion
        HTTPS = "https://"
        if mr_url.startswith(HTTPS):
            mr_url = mr_url[len(HTTPS):]
```